datadog/dogshell/service_level_objective.py

```diff
@@ -0,0 +1,418 @@
+        slo = json.load(args.file)
+
+        params = {
+            "type": slo["type"],
+            "name": slo["name"],
+            "thresholds": slo["thresholds"],
+        }
+
+        if slo.get("description"):
+            params["description"] = slo["description"]
+
+        if slo["type"] == "metric":
+            params["query"] = {
+                "numerator": slo["numerator"],
+                "denominator": slo["denominator"],
+            }
+        elif slo.get("monitor_search"):
+            params["monitor_search"] = slo["monitor_search"]
+        else:
+            params["monitor_ids"] = slo["monitor_ids"]
+
+        if slo.get("tags"):
+            tags = sorted(set([t.strip() for t in slo["tags"].split(",") if t.strip()]))
+            params["tags"] = tags
+
+        res = api.ServiceLevelObjective.create(**params)
+        report_warnings(res)
+        report_errors(res)
+        if format == "pretty":
+            print(pretty_json(res))
+        else:
+            print(json.dumps(res))
+
+    @classmethod
+    def _update(cls, args):
+        api._timeout = args.timeout
+        format = args.format
+
+        params = {"type": args.type}
+
+        if args.thresholds:
+            thresholds = []
+            for threshold_str in args.thresholds.split(","):
+                parts = threshold_str.split(":")
+                timeframe = parts[0]
+                target = parts[1]
+                warning = None
+                if len(parts) > 2:
+                    warning = parts[2]
+                thresholds.append(
+                    {"timeframe": timeframe, "target": target, "warning": warning}
+                )
+            params["thresholds"] = thresholds
+
+        if args.description:
+            params["description"] = args.description
+
+        if args.type == "metric":
+            if args.numerator and args.denominator:
+                params["query"] = {
+                    "numerator": args.numerator,
+                    "denominator": args.denominator,
+                }
+        elif args.monitor_search:
+            params["monitor_search"] = args.monitor_search
+        else:
+            params["monitor_ids"] = args.monitor_ids
+
+        if args.tags:
+            tags = sorted(set([t.strip() for t in args.tags.split(",") if t.strip()]))
+            params["tags"] = tags
+        res = api.ServiceLevelObjective.update(args.slo_id, **params)
+        report_warnings(res)
+        report_errors(res)
+        if format == "pretty":
+            print(pretty_json(res))
+        else:
+            print(json.dumps(res))
+
+    @classmethod
+    def _file_update(cls, args):
+        api._timeout = args.timeout
+        format = args.format
+        slo = json.load(args.file)
+
+        params = {"type": slo["type"], "name": slo["name"]}
+
+        if slo.get("thresholds"):
+            thresholds = []
+            for threshold_str in args.thresholds.split(","):
```

Comment:
same here, thresholds should already be in a proper JSON format, like you did in `file_create`